packages/asc-ui/src/components/Radio/Radio.tsx

```diff
@@ -1,27 +1,85 @@
-import * as React from 'react'
-import RadioItemStyle from './RadioItemStyle'
-import RadioLabelStyle from './RadioLabelStyle'
-import RadioGroupStyle from './RadioGroupStyle'
+import React, { useState, useContext, useMemo } from 'react'
+import RadioStyle, {
+  RadioWrapperStyle,
+  RadioCircleStyle,
+  Props,
+} from './RadioStyle'
+import RadioContext from './RadioContext'
+import LabelContext from '../Label/LabelContext'
 
-interface Props {
-  id: string
-  name?: string
-  label: string
-  value?: string
-  defaultChecked?: boolean
-  disabled?: boolean
-}
+const Radio: React.FC<Props & React.HTMLAttributes<HTMLDivElement>> = ({
+  className,
+  onChange,
+  variant,
+  disabled,
+  id,
+  defaultChecked,
+  error: errorProp,
+  ...otherProps
+}) => {
+  const [focus, setFocus] = useState(false)
+  const {
+    setChecked,
+    checked: checkedProp,
+    name,
+    error: errorGroup,
+  } = useContext(RadioContext)
+  const { setActive } = useContext(LabelContext)
+  const checked = checkedProp === id
+  const error = errorProp || errorGroup || false
 
-const Radio = ({ id, value, label, disabled }: Props) => (
-  <RadioItemStyle>
-    <RadioLabelStyle htmlFor={id} disabled={disabled}>
-      {label || value}
-    </RadioLabelStyle>
-  </RadioItemStyle>
-)
+  // Make the label aware of changes in the checked state
+  useMemo(() => {
+    setActive(checked)
+  }, [checked, setActive])
```

Comment:
Should `setActive` be a dependency here? It's a function and should not change, should it?

Reply:
@chrisvanmook has a reason for this

Reply:
Checked this; has to do with the linter rules and the React core team recommendations on how dependencies should be used.